src/core/server/saved_objects/serialization/serializer.ts

```diff
@@ -59,6 +59,9 @@ export class SavedObjectsSerializer {
     doc: SavedObjectsRawDoc,
     options: SavedObjectsRawDocParseOptions = {}
   ): SavedObjectSanitizedDoc {
+    if (!this.isRawSavedObject(doc, options)) {
+      throw new Error(`Raw document '${doc._id}' is not a valid saved object`);
```

Comment:
Maybe we should show a reason for the error? There are several possible reasons: SO `_source.type` is not defined, id prefix doesn't match, probably we will have more in the future.

Reply:
Good idea, done in cfa3520e4.